cartridge-cli.lua

```diff
@@ -3505,82 +3590,88 @@ function Process:initialize()
     self.env.TARANTOOL_CONSOLE_SOCK = self.console_sock
 end
 
-function Process:check_pid_file()
+function Process:is_running()
     if fio.stat(self.pid_file) then
         local pid = tonumber(utils.read_file(self.pid_file))
         if pid == nil or pid <= 0 then
-            error('Pid file exists with unknown format: ' .. self.pid_file)
+            return nil, string.format('Pid file exists with unknown format: %s', self.pid_file)
         elseif check_pid_running(pid) then
-            error('Process is already running with pid_file: ' .. self.pid_file)
-        else
-            assert(fio.unlink(self.pid_file))
+            return true
         end
     end
 end
 
-function Process:start_in_foreground()
-    utils.write_file(self.pid_file, require('tarantool').pid(), tonumber('644', 8))
-    execve(arg[-1], {self.script}, self.env) -- stops execution
-end
-
 function Process:build_notify_socket()
-    local sock = assert(socket('AF_UNIX', 'SOCK_DGRAM', 0), 'Can not create socket')
+    local sock = socket('AF_UNIX', 'SOCK_DGRAM', 0)
+    if sock == nil then return nil, 'Cannot create notify socket' end
+
     local basename = self.app_name .. '.' .. self.instance_name .. '-notify.sock'
     local sock_name = fio.pathjoin(self.run_dir, basename)
```

Comment:
Please, verify socket length - https://github.com/tarantool/tarantool/issues/4634

Reply:
For notify socket: https://github.com/tarantool/cartridge-cli/issues/214
For instance socket: https://github.com/tarantool/cartridge/issues/790